go-controller/pkg/ovn/port_cache.go

```diff
@@ -37,7 +37,8 @@ func (c *portCache) get(logicalPort string) (*lpInfo, error) {
 	c.RLock()
 	defer c.RUnlock()
 	if info, ok := c.cache[logicalPort]; ok {
-		return info, nil
+		x := *info
```

Comment:
lol :) I remember you mentioned we were unsafely returning the original pointer instead of a copy. just want to make sure we didn't actually have a reason for doing so? like do we have some logic somewhere we want to change the actual struct after the get?

Reply:
i looked around and I didn't see any usage like that.

Reply:
ack, let's stick to this safer way.

Reply:
No it's not the same. See: https://go.dev/play/p/XkkNoeaf5vW